### PR TITLE
Reusable Blocks: Change from store name provided as a string to store definition

### PR DESCRIPTION
**Pull request by grzim**
## Description
Regarding https://github.com/WordPress/gutenberg/issues/27088 
Change from store name provided as a string to store definition applied

## How has this been tested?
Not tested yet

## Types of changes 
Code refactor (non-breaking change)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/reusable-blocks/src/store/controls.js [hunk 1]
```diff
@@ -10,6 +10,11 @@ import {
 import { createRegistryControl } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
+/**
+ * Internal dependencies
+ */
+import { store as reusableBlocksStore } from './index.js';
```
Comment (gziolo) [inline packages/reusable-blocks/src/store/controls.js:16]: This is surprising to see a cyclic dependency between `store/index.js` and `store/controls.js`. Would it be possible to access somehow `dispatch` from the current store instead? /cc @adamziel

**Comment by gziolo** [comment]
It looks like we can now remove the following statements from some packages:
```js
import '@wordpress/reusable-blocks';
```

Screen Shot 2020-11-19 at 14 53 39

@adamziel, is there anything I could miss that justifies those imports after the refactoring proposed?

**Comment by gziolo** [comment]
@grzim, thank you for opening this PR. In general, this looks good, I had some question to @adamziel who according to git authored `@wordpress/reusable-blocks` package.

File: packages/reusable-blocks/src/store/controls.js (hunk 1 above)
Comment (adamziel) [reply packages/reusable-blocks/src/store/controls.js:16]: @gziolo The control needs to dispatch data on that store so it's not easily avoidable. I think the only way of avoiding circular dependency would be not refactoring this file and just leaving "core/reusable-blocks" string as it was before. This is going to be a general theme in plenty of registry actions and selectors too - we should address all those cases in the same way and either accept the circular dependency (it shouldn't break anything in this case) or use string to refer to the store from the same package.

Comment (gziolo) [reply packages/reusable-blocks/src/store/controls.js:16]: True, reusing the constant sounds like a good way of handling this particular case 👍

**Review by gziolo — APPROVED** [review]
This PR is good, we can always open a follow-up to address the usage in controls.

**Pull request merged**